builtin/providers/aws/resource_aws_glacier_vault.go

```diff
@@ -54,9 +54,13 @@ func resourceAwsGlacierVault() *schema.Resource {
 			},
 
 			"access_policy": &schema.Schema{
-				Type:      schema.TypeString,
-				Optional:  true,
-				StateFunc: normalizeJson,
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
```

Comment:
We should replace this with the diff policy helper we use elsewhere, but it's unrelated to the original purpose of this PR, so I can submit a separate PR for it.